core/arch/arm/kernel/boot.c

```diff
@@ -1621,3 +1627,89 @@ unsigned long __weak get_aslr_seed(void *fdt __unused)
 }
 #endif /*!CFG_DT*/
 #endif /*CFG_CORE_ASLR*/
+
+#if defined(CFG_CORE_SEL2_SPMC) && defined(CFG_CORE_RELOCATABLE)
+static void *get_fdt_from_boot_info(struct ffa_boot_info_header *hdr)
+{
+	struct ffa_boot_info *desc = NULL;
+	uint8_t content_fmt = 0;
+	uint8_t name_fmt = 0;
+	void *fdt = NULL;
+	int ret = 0;
+
+	if (hdr->signature != FFA_BOOT_INFO_SIGNATURE) {
+		EMSG("Bad boot info signature %#"PRIx32, hdr->signature);
+		panic();
+	}
+	if (hdr->version != FFA_BOOT_INFO_VERSION) {
+		EMSG("Bad boot info version %#"PRIx32, hdr->version);
+		panic();
+	}
+	if (hdr->desc_count != 1) {
+		EMSG("Bad boot info descriptor count %#"PRIx32,
+		     hdr->desc_count);
+		panic();
+	}
+	desc = (void *)((vaddr_t)hdr + hdr->desc_offset);
```

Comment:
depending on the desc_offset desc might end up being unaligned, do we care here?

Reply:
I don't think we care or we should check the alignment of `hdr` also. We could perhaps add a few asserts to help debugging, but I'm not sure it's worth it.